crates/sui-json-rpc-api/src/lib.rs

```diff
@@ -237,7 +237,63 @@ impl TransactionBytes {
 
 #[open_rpc(namespace = "sui", tag = "Event Subscription")]
 #[rpc(server, client, namespace = "sui")]
-pub trait EventApi {
+pub trait EventStreamingApi {
     #[subscription(name = "subscribeEvent", item = SuiEventEnvelope)]
     fn subscribe_event(&self, filter: SuiEventFilter);
 }
+
+#[open_rpc(namespace = "sui", tag = "Event Read API")]
+#[rpc(server, client, namespace = "sui")]
+pub trait EventReadApi {
+    #[method(name = "getEventsByTransaction")]
+    async fn get_events_by_transaction(
+        &self,
+        digest: TransactionDigest,
+    ) -> RpcResult<Vec<SuiEventEnvelope>>;
+
+    #[method(name = "getEventsByModule")]
+    async fn get_events_by_module(
+        &self,
+        package: ObjectID,
+        module: String,
+        count: u64,
+        start_time: u64,
+        end_time: u64,
+    ) -> RpcResult<Vec<SuiEventEnvelope>>;
+
+    #[method(name = "getEventsByEventType")]
+    async fn get_events_by_event_type(
+        &self,
+        event_type: String,
```

Comment:
Is this the `EventType` (ie TransferObject, etc.) or the Move module/package etc?

Reply:
this is the `EventType` e.g. `TransferObject` or `0x2::my_module::my_event`. 
The other package/.module thing is the location where the events are emitted.